packages/hdwallet-tallyho/src/adapter.ts

```diff
@@ -0,0 +1,88 @@
+import * as core from "@shapeshiftoss/hdwallet-core";
+import TallyHoOnboarding from "tallyho-onboarding";
+
+import { TallyHoHDWallet } from "./tallyho";
+
+interface TallyHoEthereumProvider {
+  isTally?: boolean;
+}
+
+interface Window {
+  ethereum?: TallyHoEthereumProvider;
+}
+
+export class TallyHoAdapter {
+  keyring: core.Keyring;
+
+  private constructor(keyring: core.Keyring) {
+    this.keyring = keyring;
+  }
+
+  public static useKeyring(keyring: core.Keyring) {
+    return new TallyHoAdapter(keyring);
+  }
+
+  public async pairDevice(): Promise<TallyHoHDWallet> {
+    const provider: any = await this.detectTallyProvider();
+    if (!provider) {
+      const onboarding = new TallyHoOnboarding();
+      onboarding.startOnboarding();
+      console.error("Please install Tally Ho!");
+    }
+    if (provider === null) {
+      throw new Error("Could not get Tally Ho accounts.");
+    }
+
+    // eslint-disable-next-line no-useless-catch
+    try {
+      await provider.request({ method: "eth_requestAccounts" });
+    } catch (error) {
+      throw error;
+    }
+    const wallet = new TallyHoHDWallet(provider);
+    const deviceID = await wallet.getDeviceID();
+    this.keyring.add(wallet, deviceID);
+    this.keyring.emit(["Tally Ho", deviceID, core.Events.CONNECT], deviceID);
+
+    return wallet;
+  }
+
+  public async detectTallyProvider(): Promise<TallyHoEthereumProvider | null> {
```

Comment:
this should be private. no one outside of this class should be calling this. it's not part of the public interface.